hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/protocolPB/OzoneManagerRequestHandler.java

```diff
@@ -416,6 +471,32 @@ private ListBucketsResponse listBuckets(ListBucketsRequest request)
     return resp.build();
   }
 
+  @RequestFeatureValidator(
+      conditions = ValidationCondition.OLDER_CLIENT_REQUESTS,
+      processingPhase = RequestProcessingPhase.POST_PROCESS,
+      requestType = Type.ListBuckets
+  )
+  public static OMResponse disallowListBucketsResponseWithECReplicationConfig(
```

Comment:
Why disallow the list bucket command? Ozone CLI can be an older version and it should be able to list all the buckets.

Reply:
As we discussed in the JIRA and offline as well, there are two things we can do here:
- filter EC replicated stuff from the result
- fail the request

If we filter, there can be two inconsistent views of the filesystem from an old and from a new client (which might be on the same user machine in two different softwares), which can lead to confusion. So it seems to be safer to just fail the request.